packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -228,89 +215,12 @@ export class PendingStateManager implements IDisposable {
-    }
-
-    /**
-     * Listens for ACKs of stashed ops
-     */
-    private processRemoteMessage(message: ISequencedDocumentMessage) {
-        if (!isRuntimeMessage(message)) {
-            return { localAck: false, localOpMetadata: undefined };
-        }
-
-        // this message was a pending op that was actually sent successfully
-        const isOriginalClientId = message.clientId === Array.from(this.previousClientIds)[0] &&
-            message.clientSequenceNumber >= this.firstStashedCSN;
-        // this message is a pending or stashed op that was resubmitted
-        const isNewClientId = Array.from(this.previousClientIds).indexOf(message.clientId) > 0;
-
-        // if this is an ack for a stashed op, dequeue one message.
-        // we should have seen its ref seq num by now and the DDS should be ready for it to be ACKed
-        if (isOriginalClientId || isNewClientId) {
-            assert(this.clientId === undefined, 0x28b /* "multiple clients connected with stashed ops" */);
-            while (!this.pendingStates.isEmpty()) {
-                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-                const nextState = this.pendingStates.shift()!;
-                // if it's not a message just drop it and keep looking
-                if (nextState.type === "message") {
-                    this.assertOpMatch(nextState, message, isOriginalClientId);
-                    return { localAck: true, localOpMetadata: nextState.localOpMetadata };
-                }
-            }
-        }
-
-        if (message.type === ContainerMessageType.Rejoin && this.previousClientIds.has(message.contents?.clientId)) {
-            this.previousClientIds.add(message.clientId);
-        }
-
-        return { localAck: false, localOpMetadata: undefined };
-    }
-
-    private assertOpMatch(state: IPendingMessage, message: ISequencedDocumentMessage, isOriginalClientId: boolean) {
-        assert(message.type === state.messageType, 0x28c /* "different message type" */);
-        assert(message.clientSequenceNumber === state.clientSequenceNumber || !isOriginalClientId,
-            0x28d /* "client sequence number doesn't match" */);
-        switch(message.type) {
-            case ContainerMessageType.Attach:
-                assert(message.contents.id === state.content.id, 0x28e /* "datastore ID doesn't match" */);
-                break;
-            case ContainerMessageType.FluidDataStoreOp:
-                assert(message.contents.address === state.content.address, 0x28f /* "address doesn't match" */);
-                break;
-            case ContainerMessageType.BlobAttach:
-                // todo: assert we have blob storage, assert blob IDs match, remove blob from blob storage since it made
-                // it through successfully
-                break;
-            case ContainerMessageType.Rejoin:
-            default:
-                throw new Error(`${message.type} not expected`);
-        }
```

Comment:
Are these asserts no longer relevant?

Reply:
I added these when I added the rejoin op and I don't think they're really necessary now. It wouldn't hurt to leave them in, though they won't necessarily catch mismatched ops (e.g., multiple ops to the same data store)